numpy/fft/tests/test_pocketfft.py

```diff
@@ -465,6 +465,28 @@ def test_irfftn_out_and_s_interaction(self, s):
         assert result is out
         assert_array_equal(result, expected)
 
+    def test_fftn_with_out_and_s(self):
```

Comment:
I'd suggest to use a `pytest.mark.parametrize` here to do the two tests and add more possible `s` tuples, making values both bigger and smaller than the array in various positions. Also, could you move the test right below `test_fftn_out_and_s_interaction` (and maybe call it `test_fftn_out_and_s_interaction2`).